dogfood.py

```diff
@@ -0,0 +1,145 @@
+
+    url = "https://api.github.com/repos/DataDog/" + repository + "/pulls"
+    response = requests.post(url=url, headers=headers, data=data)
+    if response.status_code == 201:
+        print("✔ Pull Request created successfully")
+        return 0
+    else:
+        print("✘ pull request failed " + str(response.status_code) + '\n' + response.text)
+        return 1
+
+
+def generate_target_code(target: str, temp_dir_path: str, version: str):
+    print("… Generating code with version " + version)
+
+    if target == TARGET_APP:
+        target_file = os.path.join(temp_dir_path, "Podfile")
+        content = ""
+        with open(target_file, 'r') as target:
+            lines = target.readlines()
+            for line in lines:
+                if "pod 'DatadogSDK'" in line:
+                    content = content + "    pod 'DatadogSDK', :git => 'https://github.com/DataDog/dd-sdk-ios.git', :tag => '" + version + "'\n"
+                else:
+                    content = content + line
+
+        with open(target_file, 'w') as target:
+            target.write(content)
+
+        target_file = os.path.join(temp_dir_path, "Podfile.lock")
+        content = ""
+        with open(target_file, 'r') as target:
+            lines = target.readlines()
+            for line in lines:
+                if "  - DatadogSDK (from " in line:
+                    content = content + "  - DatadogSDK (from `https://github.com/DataDog/dd-sdk-ios.git`, tag `" + version + "`)\n"
+                elif "  - DatadogSDK (" in line:
+                    content = content + "  - DatadogSDK (" + version + ")\n"
+                else:
+                    content = content + line
+
+        with open(target_file, 'w') as target:
+            target.write(content)
+
+    # TODO RUMM-1063 elif target == TARGET_DEMO: …
+
+
+def git_clone_repository(repo_name: str, gh_token: str, temp_dir_path: str) -> Tuple[Repo, str]:
+    print("… Cloning repository " + repo_name)
+    url = "https://" + gh_token + ":x-oauth-basic@github.com/DataDog/" + repo_name
+    repo = Repo.clone_from(url, temp_dir_path)
+    base_name = repo.active_branch.name
+    return repo, base_name
+
+
+def git_push_changes(repo: Repo, version: str):
+    print("… Committing changes")
+    repo.git.add(update=True)
+    repo.index.commit("Update DD SDK to " + version)
+
+    print("⑊ Pushing branch")
+    origin = repo.remote(name="origin")
+    repo.git.push("--set-upstream", "--force", origin, repo.head.ref)
+
+
+def update_dependant(version: str, target: str, gh_token: str) -> int:
+    branch_name = "update_sdk_" + version
+    temp_dir = TemporaryDirectory()
+    temp_dir_path = temp_dir.name
+    repo_name = REPOSITORIES[target]
+
+    repo, base_name = git_clone_repository(repo_name, gh_token, temp_dir_path)
+
+    print("… Creating branch " + branch_name)
+    repo.git.checkout('HEAD', b=branch_name)
+
+    generate_target_code(target, temp_dir_path, version)
```

Comment:
I actually added code to update the Podfile.lock directly in the script. But we can try to run `bundle exec pod install` indeed

Reply:
We will need to run `bundle exec pod install` indeed since you update only the version on the `Podfile.lock` file but a hash is computed at the end of the file to ensure the lock file wasn't edited directly by an external script.